src/verification/index.ts

```diff
@@ -2,46 +2,134 @@
  * @file Manages verification of otp form fields (email, sms, whatsapp)
  * @author Jean Tan
  */
-import { PackageInitParams } from '../types'
+import nacl from 'tweetnacl'
+import { decodeUTF8, decodeBase64, encodeBase64 } from 'tweetnacl-util'
 
-import authenticate from './authenticate'
-import generateSignature from './generate-signature'
-import getPublicKey from './get-public-key'
+import {
+  VerificationSignatureOptions,
+  VerificationAuthenticateOptions,
+} from '../types'
 
-/**
- * Provider that accepts configuration
- * before returning the webhooks module
- */
-export = function ({
-  mode,
-  verificationOptions,
-  publicKey,
-}: PackageInitParams) {
-  if (verificationOptions !== undefined) {
-    const verificationPublicKey =
-      publicKey || getPublicKey(mode || 'production')
-    const {
-      secretKey: verificationSecretKey,
-      transactionExpiry,
-    } = verificationOptions
-    return {
-      authenticate:
-        transactionExpiry !== undefined
-          ? authenticate(verificationPublicKey, transactionExpiry)
-          : function () {
-              throw new Error(
-                'Provide transactionExpiry when initializing the formsg sdk to use this function.'
-              )
-            },
-      generateSignature:
-        verificationSecretKey !== undefined
-          ? generateSignature(verificationSecretKey)
-          : function () {
-              throw new Error(
-                'Provide verificationSecretKey when initializing the formsg sdk to use this function.'
-              )
-            },
+import { MissingSecretKeyError, MissingPublicKeyError } from '../errors'
+import { parseVerificationSignature } from '../util/parser'
+
+import { isSignatureTimeValid, formatToBaseString } from './utils'
+
+export default class Verification {
+  verificationPublicKey?: string
+  verificationSecretKey?: string
+  transactionExpiry?: number
+
+  constructor({
+    verificationPublicKey,
+    verificationSecretKey,
+    transactionExpiry,
+  }: {
+    verificationPublicKey?: string
+    verificationSecretKey?: string
+    transactionExpiry?: number
+  }) {
+    this.verificationPublicKey = verificationPublicKey
+    this.verificationSecretKey = verificationSecretKey
+    this.transactionExpiry = transactionExpiry
+  }
+
+  /**
+   *  Verifies signature
+   * @param {object} data
+   * @param {string} data.signatureString
+   * @param {number} data.submissionCreatedAt date in milliseconds
+   * @param {string} data.fieldId
+   * @param {string} data.answer
+   * @param {string} data.publicKey
+   */
+  authenticate = ({
+    signatureString,
+    submissionCreatedAt,
+    fieldId,
+    answer,
+  }: VerificationAuthenticateOptions) => {
+    if (!this.transactionExpiry) {
```

Comment:
wouldn't it be better to move these checks into the constructor? similarly for `this.verificationPublicKey` and `this.verificationSecretKey`

Reply:
The plan is to add flexibility to the user -- one missing config should not prevent the user from using other functions that do not require the missing config. 

Is that _too_ flexible?

Reply:
The other method requires a verification secret key, and not a public key :P